Makefile

```diff
@@ -3,13 +3,53 @@
 SOURCES = AddressResolver.cpp Profile.cpp
 HEADERS = AddressResolver.h Profile.h
 
+KCACHEGRIND = kcachegrind
+
+CFLAGS = -Wall -Wextra -g
+PGCOLLECT_FLAGS = -F 2048 -s
+
 all: pgcollect pginfo pgconvert
 
 pgcollect: pgcollect.c
-	$(CC) -std=gnu99 -Wall -Wextra -g -D_GNU_SOURCE -o pgcollect pgcollect.c ${FLAGS}
+	$(CC) -std=gnu99  -O2 $(CFLAGS) ${FLAGS} -D_GNU_SOURCE -o pgcollect  pgcollect.c
+
+pgconvert: pgconvert.cpp $(SOURCES) $(HEADERS)
+	$(CXX) -std=c++03 -O2 $(CFLAGS) ${FLAGS}               -o pgconvert  pgconvert.cpp $(SOURCES) -ldw -lelf
 
 pginfo: pginfo.cpp $(SOURCES) $(HEADERS)
-	$(CXX) -std=c++03 -Wall -Wextra -g -o pginfo pginfo.cpp $(SOURCES) -ldw -lelf ${FLAGS}
+	$(CXX) -std=c++03 -O2 $(CFLAGS) ${FLAGS}               -o pginfo     pginfo.cpp    $(SOURCES) -ldw -lelf
 
-pgconvert: pgconvert.cpp $(SOURCES) $(HEADERS)
-	$(CXX) -std=c++03 -Wall -Wextra -g -o pgconvert pgconvert.cpp $(SOURCES) -ldw -lelf ${FLAGS}
+# only used to be traced itself
+pginfo_dbg: pginfo.cpp $(SOURCES) $(HEADERS)
+	$(CXX) -std=c++03 -O0 $(CFLAGS) ${FLAGS}               -o pginfo_dbg pginfo.cpp    $(SOURCES) -ldw -lelf
```

Comment:
If we leave it for now, we need -fno-omit-frame-pointer here

Reply:
I've gone with  `-O` (needed for some system CFLAGS that include things like `-DFORTIFY_SOURCE`) and with `-ggdb3` which leaves a lot of debug info in.